portal-impl/src/com/liferay/portal/repository/liferayrepository/LiferayFileEntryValidatorLocalRepositoryWrapper.java

```diff
@@ -0,0 +1,152 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2025 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.portal.repository.liferayrepository;
+
+import com.liferay.document.library.kernel.model.DLVersionNumberIncrease;
+import com.liferay.document.library.kernel.util.DLValidator;
+import com.liferay.portal.kernel.exception.PortalException;
+import com.liferay.portal.kernel.repository.LocalRepository;
+import com.liferay.portal.kernel.repository.model.FileEntry;
+import com.liferay.portal.kernel.service.ServiceContext;
+import com.liferay.portal.kernel.util.Validator;
+import com.liferay.portal.repository.util.LocalRepositoryWrapper;
+import com.liferay.portlet.documentlibrary.util.DLAppUtil;
+
+import java.io.File;
+import java.io.InputStream;
+
+import java.util.Date;
+
+/**
+ * @author Alicia García
+ */
+public class LiferayFileEntryValidatorLocalRepositoryWrapper
+	extends LocalRepositoryWrapper {
+
+	public LiferayFileEntryValidatorLocalRepositoryWrapper(
+		LocalRepository localRepository, DLValidator dlValidator) {
+
+		super(localRepository);
+
+		_dlValidator = dlValidator;
+	}
+
+	@Override
+	public FileEntry addFileEntry(
+			String externalReferenceCode, long userId, long folderId,
+			String sourceFileName, String mimeType, String title,
+			String urlTitle, String description, String changeLog, File file,
+			Date displayDate, Date expirationDate, Date reviewDate,
+			ServiceContext serviceContext)
+		throws PortalException {
+
+		long size = 0;
+
+		if (file != null) {
+			size = file.length();
+		}
```

Comment:
Suggestion: can we use a conditional expression here? `long size = (file == null) ? 0 : file.length()`